core-bundle/src/Routing/ResponseContext/CoreResponseContextFactory.php

```diff
@@ -98,9 +101,11 @@ public function createContaoWebpageResponseContext(PageModel $pageModel): Respon
 
             // Ensure absolute links
             if (!preg_match('#^https?://#', $url)) {
-                /** @var Environment $environment */
-                $environment = $this->contaoFramework->getAdapter(Environment::class);
-                $url = $environment->get('base').$url;
+                if (!$request = $this->requestStack->getMainRequest()) {
+                    throw new \RuntimeException('The request stack did not contain a request');
+                }
+
+                $url = $request->getSchemeAndHttpHost().'/'.$url;
```

Comment:
That's not correct, is it? Our base handling is not just the current request which is precisely why I did not use the request stack. Our "base" is what's in `<base>` and this might also include the subfolder.

Reply:
But `Environment::get('base')` is made up of `return static::get('url') . static::get('path') . '/';`, which translates to `$request->getSchemeAndHttpHost()` and `$request->getBasePath()`. And since the base path is already prepended when the insert tag is replaced (probably with an existing subfolder?), only `$request->getSchemeAndHttpHost()` remains. 🤔

Reply:
I don't know but if you say so :D

Reply:
I wonder why the base path is already there, though. PageModel::getFrontendUrl() does not add it actually.